include/nuttx/arch.h

```diff
@@ -1353,6 +1354,39 @@ void up_irqinitialize(void);
 
 bool up_interrupt_context(void);
 
+/****************************************************************************
+ * Name: up_irq_save
+ *
+ * Description:
+ *   Save the current interrupt state and disable interrupts.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   Interrupt state prior to disabling interrupts.
+ *
+ ****************************************************************************/
+
+irqstate_t up_irq_save(void);
+
+/****************************************************************************
+ * Name: up_irq_restore
+ *
+ * Description:
+ *   Restore the previous irq state (i.e., the one previously
+ *   returned by up_irq_save())
+ *
+ * Input Parameters:
+ *   irqstate - The interrupt state to be restored.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void up_irq_restore(irqstate_t irqstate);
+
```

Comment:
@btashton @v01d Building against 10.1 the irq.h defines `up_irq_restore` and `up_irq_save` as inlines, but these definition are not
```
nuttx/include/nuttx/arch.h:1386:12: error: conflicting declaration of 'irqstate_t up_irq_save()' with 'C' linkage
 1386 | irqstate_t up_irq_save(void);
      |            ^~~~~~~~~~~
```

Should this not have come in?

Reply:
I guess we missed this during review. From the looks of it this declaration should only be in arch-dependant header, as it is done for other arch. You can make a PR to remove these and we can see if BL602 still builds or it requires further fix.

Reply:
Sorry I am missing the issue here what are you trying to build. The bl602 configurations build fine.

Reply:
This added a declaration in system-wide header, which conflicts with arch-dependant definitions (which in some cases as inline). It probably affects other arch's.

Reply:
Strange I was not seeing it in the patch but now I do. My question is the same though we're are you seeing this come up? It's been building cleanly for months.

Reply:
I know, very strange, it could be the build setting under PX4 not allowing the warning. But it is fundamentally wrong.  

https://github.com/apache/incubator-nuttx/pull/3580